packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -176,11 +184,63 @@ contract BridgePool is Testable {
         );
     }
 
-    function speedUpRelay(uint256 depositId) public {}
+    function speedUpRelay(bytes32 _relayHash, RelayAncillaryDataContents memory _relayData) public {
+        verifyRelayHash(_relayHash, _relayData);
+    }
+
+    function finalizeRelay(bytes32 _relayHash, RelayAncillaryDataContents memory _relayData) public {
+        verifyRelayHash(_relayHash, _relayData);
+    }
+
+    function settleDisputedRelay(bytes32 _relayHash, RelayAncillaryDataContents memory _relayData) public {
+        verifyRelayHash(_relayHash, _relayData);
+    }
+
+    /**
+     * @notice OptimisticOracle will callback to this function after a pending relay is disputed.
+     */
+    function priceDisputed(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        uint256 refund
+    ) public {
+        Deposit storage pendingDeposit = deposits[ancillaryData];
+        // TODO: Implement logic. For now, placeholder code demonstrates how relay can be migrated from pending to
+        // disputed mapping.
+        Deposit storage disputedDeposit = disputedDeposits[ancillaryData][msg.sender];
```

Comment:
I may be missing something, but would it work just to leave this in deposits and just store the disputer in the existing struct?

Reply:
Hm, doesn't that make it trickier to enable another relayer to come in after a relay has been disputed? cc @chrismaree